api/hypershift/v1alpha1/nodepool_types.go

```diff
@@ -901,7 +901,7 @@ type AzureNodePoolPlatform struct {
 	//
 	// +kubebuilder:default:=default
 	// +kubebuilder:validation:Required
-	SubnetName string `json:"subnetName"`
+	SubnetID string `json:"subnetID"`
```

Comment:
Can we add the restrictions and assumptions about this?

* Can it be a different subnet than the subnet of the parent cluster?
* Does the subnet need to belong to the same Azure Subscription as the parent cluster?
* Can the Azure Resource Group of the subnet be any resource group, or does it need to be some from the parent cluster?
* Can the VNet be any VNet or does it need to be the one specified in the parent cluster during creation?
* Does the azure location need to be the one from the parent cluster?

Reply:
Updated, though I didn't add the resource group info. I added language that the VNET must be the same under the same subscription id as the hosted cluster spec. We've established the VNET can exist under a different resource group but must be under the same subscription id. The parent resource group is the managed resource group so it would seem a bit confusing to me from a UX perspective to mention it here.

Location isn't an input in creating the subnet through the CLI; doesn't the VNET location drive the subnet location?

Reply:
>Location isn't an input in creating the subnet through the CLI; doesn't the VNET location drive the subnet location?

it inherits the vnet location